ledger/acctupdates_test.go

```diff
@@ -881,30 +883,157 @@ func TestAcctUpdatesDeleteStoredCatchpoints(t *testing.T) {
-		f, err := os.Create(fmt.Sprintf("./dummy_catchpoint_file-%d", i))
+		file := fmt.Sprintf("./%v/%v/%v/dummy_catchpoint_file-%d", catchpointDir, i/10, i/2, i)
+		dummyCatchpointFiles[i] = file
+		err := os.MkdirAll(path.Dir(file), 0755)
+		require.NoError(t, err)
+		f, err := os.Create(file)
 		require.NoError(t, err)
 		err = f.Close()
 		require.NoError(t, err)
-	}
-
-	for i := 0; i < dummyCatchpointFilesToCreate; i++ {
-		err := au.accountsq.storeCatchpoint(context.Background(), basics.Round(i), fmt.Sprintf("./dummy_catchpoint_file-%d", i), "", 0)
+		err = au.accountsq.storeCatchpoint(context.Background(), basics.Round(i), file, "", 0)
 		require.NoError(t, err)
 	}
+
+	defer func() {
+		os.RemoveAll(catchpointDir)
+	}()
+
 	err = au.deleteStoredCatchpoints(context.Background(), au.accountsq)
 	require.NoError(t, err)
 
-	for i := 0; i < dummyCatchpointFilesToCreate; i++ {
-		// ensure that all the files were deleted.
-		_, err := os.Open(fmt.Sprintf("./dummy_catchpoint_file-%d", i))
+	// ensure that all the files were deleted.
+	for _, file := range dummyCatchpointFiles {
+		_, err := os.Open(file)
 		require.True(t, os.IsNotExist(err))
 	}
+
 	fileNames, err := au.accountsq.getOldestCatchpointFiles(context.Background(), dummyCatchpointFilesToCreate, 0)
 	require.NoError(t, err)
 	require.Equal(t, 0, len(fileNames))
+
+	files, err := ioutil.ReadDir(catchpointDir)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(files))
+}
+
+// The test validate that when algod boots up it cleans empty catchpoint directories.
+// it is done be creating empty directories in the catchpoint root directory.
+// When algod boots up it should remove those directories
+func TestSchemaUpdateDeleteStoredCatchpoints(t *testing.T) {
+	proto := config.Consensus[protocol.ConsensusCurrentVersion]
+	const catchpointDir string = "./catchpoints"
+
+	// creating empty catchpoint directories
+	emptyDirPath := path.Join(catchpointDir, "2f", "e1")
+	err := os.MkdirAll(emptyDirPath, 0755)
+	require.NoError(t, err)
+	emptyDirPath = path.Join(catchpointDir, "2e", "e1")
+	err = os.MkdirAll(emptyDirPath, 0755)
+	require.NoError(t, err)
+	emptyDirPath = path.Join(catchpointDir, "14", "2e", "e1")
+	err = os.MkdirAll(emptyDirPath, 0755)
+	require.NoError(t, err)
+
+	// creating catchpoint file
+	catchpointFilePath := path.Join(catchpointDir, "14", "2e", "e4", "dummy_catchpoint_file")
+	err = os.MkdirAll(path.Dir(catchpointFilePath), 0755)
+	require.NoError(t, err)
+	f, err := os.Create(catchpointFilePath)
+	require.NoError(t, err)
+	f.Close()
+	defer func() {
+		os.RemoveAll(catchpointDir)
+	}()
+	ml := makeMockLedgerForTracker(t, true, 10, protocol.ConsensusCurrentVersion)
+	defer ml.Close()
+
+	accts := []map[basics.Address]basics.AccountData{randomAccounts(20, true)}
+	au := &accountUpdates{}
+	conf := config.GetDefaultLocal()
+	conf.CatchpointInterval = 1
+	au.initialize(conf, ".", proto, accts[0])
+	defer au.close()
+
+	err = au.loadFromDisk(ml)
```

Comment:
There is no test here to ensure the said directories were truly removed.

Reply:
oops...
Done!